resources/js/app.js

```diff
@@ -5,42 +5,83 @@
  */
 
 require('./bootstrap');
-import axios from 'axios';
+/*
+  Import Required Classes
+*/
+import Vue from 'vue'
+import VueRouter from 'vue-router'
+import ApolloClient from 'apollo-boost'
+import VueApollo from 'vue-apollo'
+import Cookie from 'js-cookie'
+import App from './views/App'
+import Passwords from './views/Passwords'
+import NotFound from './views/errors/NotFound'
+import Dashboard from './views/Dashboard'
 
-window.Vue = require('vue');
+Vue.use(VueRouter);
+Vue.use(VueApollo);
 
-/**
- * The following block of code may be used to automatically register your
- * Vue components. It will recursively scan this directory for the Vue
- * components and automatically register them with their "basename".
- *
- * Eg. ./components/ExampleComponent.vue -> <example-component></example-component>
+/*
+   Setup GraphQL Client with Authentication Headers
  */
 
-// const files = require.context('./', true, /\.vue$/i);
-// files.keys().map(key => Vue.component(key.split('/').pop().split('.')[0], files(key).default));
+const apolloClient = new ApolloClient({
+    headers: {
+        'X-XSRF-TOKEN': Cookie.get('XSRF-TOKEN'),
+        'X-Requested-With': 'XMLHttpRequest',
+        'X-CSRF-TOKEN': csrf,
+    },
+    uri: apiUri
+});
+const apolloProvider = new VueApollo({
+    defaultClient: apolloClient,
+})
 
-Vue.component(
-    'passport-clients',
-    require('./components/passport/Clients.vue').default
-);
+/*
+   Import Router Views
+ */
+
+const router = new VueRouter({
+    mode: 'history',
+    routes: [
+        {
+            path: '/',
+            name: 'dashboard',
+            component: Dashboard
+        },
+        {
+            path: '/settings/password',
+            name: 'settings.password',
+            component: Passwords,
+        },
+        {
+            path: '*',
+            component: NotFound,
+        }
+    ],
+});
+
+/*
+   Initialise Custom Components
+ */
 
 Vue.component(
```

Comment:
https://vuejs.org/v2/guide/components-registration.html#Automatic-Global-Registration-of-Base-Components - something to consider.